plugins/device/deviceutils/store-utils.go

```diff
@@ -51,14 +52,43 @@ func GetDevicesFromPeer(peerInfo *peer.Peer) ([]deviceapi.Info, error) {
 	return deviceInfo, nil
 }
 
-// DeviceInList checks whether the given device is in list of devices or not.
-func DeviceInList(reqDevice string, devices []deviceapi.Info) bool {
-	for _, key := range devices {
+// SetDeviceState sets device state and updates device state in etcd
+func SetDeviceState(peerID, deviceName, deviceState string) error {
+
+	devices, err := GetDevices(peerID)
+	if err != nil {
+		return err
+	}
+
+	index := DeviceInList(deviceName, devices)
+	if index < 0 {
+		return errors.New("device does not exists in the given peer")
```

Comment:
`does not exist`

https://ell.stackexchange.com/questions/23943/file-doesn-t-exist-or-file-doesn-t-exists